### PR TITLE
UI-Book

### PR DESCRIPTION
**Pull request by thanhconguet214**

### CODE REVIEW

**Comment by thanhconguet214** [comment]
ready

File: Gemfile [hunk 1]
```diff
@@ -1,62 +1,43 @@
...
+gem "sass-rails", "~> 5.0"
+gem "uglifier", ">= 1.3.0"
+gem "coffee-rails", "~> 4.2"
+gem "turbolinks", "~> 5"
+gem "jbuilder", "~> 2.5"
+gem "bootsnap", ">= 1.1.0", require: false
+gem "bcrypt", "3.1.12"
+gem "config"
+gem "jquery-rails"
+gem "ffaker"
```
Comment (mdmahmudurrahman) [inline Gemfile]: Em, add this gem to the development group.

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+  margin-bottom: 30px;
+  text-align: center;
+}
+
+p {
+  font-size: 1.1em;
+  line-height: 1.7em;
+}
+
+/* header */
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Don't see any CSS related to the comment

File: app/assets/stylesheets/custom.scss [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+  padding-top: 5px;
+  border-top: 1px solid $gray-medium-light;
+  display: block; 
+  small {
+    float: left;
+  }
+}
+
+.header-text{
+  color: yellow;
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Em, use color variable

File: app/assets/stylesheets/custom.scss [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+      display: block;
+      margin-bottom: 3px;
+      line-height: 1;
+    }
+    h1 {
+      font-size: 1.4em;
+      text-align: left;
+      letter-spacing: -1px;
+      margin-bottom: 3px;
+      margin-top: 0px;
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: You can combine line 112 and 111 into one

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -1,2 +1,16 @@
 module ApplicationHelper
+  def full_title(page_title = '')
```
Comment (mdmahmudurrahman) [inline app/helpers/application_helper.rb]: Please use double quote

File: app/helpers/application_helper.rb [hunk 6]
```diff
@@ -1,2 +1,16 @@
...
+    base_title = "Book FRAMGIA"
```
Comment (mdmahmudurrahman) [inline app/helpers/application_helper.rb]: I18n

File: app/helpers/details_helper.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+module DetailsHelper
```
Comment (mdmahmudurrahman) [inline app/helpers/details_helper.rb]: Please remove this unused file

File: app/helpers/static_pages_helper.rb [hunk 8]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (mdmahmudurrahman) [inline app/helpers/static_pages_helper.rb]: The same here

File: app/views/details/info.html.erb [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+                <div class="row">
+                  <h4 class="tip-left">
+                    <%= t ".branch" %>              
+                  </h4>
+                  <div class="text-warning lead"><%= t ".office" %></div>
+                </div>
+                <div class="row">
+                  <div class="check-status">
+                    <h3><%= t ".status" %></h3>
+                    <%= form_for :a, url: "#" do |f| %>
```
Comment (mdmahmudurrahman) [inline app/views/details/info.html.erb]: What is :a here ? What is the purpose ? Can you explain in brief ?

File: app/views/details/info.html.erb [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+                        <%= f.check_box :reading_me %>
+                        <span><%= t(".reading")%></span>
+                      <% end %>
+                      <%= f.label :returned, class: "checkbox inline" do %>
+                        <%= f.check_box :returned_me %>
+                        <span><%= t(".returned")%></span>
+                      <% end %>        
+                      <%= f.label :returning, class: "checkbox inline" do %>
+                        <%= f.check_box :returning_me %>
+                        <span><%= t(".returning")%></span>
```
Comment (mdmahmudurrahman) [inline app/views/details/info.html.erb]: remove optional ()

File: app/views/details/info.html.erb [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+                  <div class="check-status">
+                    <h3><%= t ".status" %></h3>
+                    <%= form_for :a, url: "#" do |f| %>
+                      <%= f.label :reading, class: "checkbox inline" do %>
+                        <%= f.check_box :reading_me %>
+                        <span><%= t(".reading")%></span>
+                      <% end %>
+                      <%= f.label :returned, class: "checkbox inline" do %>
+                        <%= f.check_box :returned_me %>
+                        <span><%= t(".returned")%></span>
```
Comment (mdmahmudurrahman) [inline app/views/details/info.html.erb]: remove optional ()

File: app/views/details/info.html.erb [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+                  </h4>
+                  <div class="text-warning lead"><%= t ".office" %></div>
+                </div>
+                <div class="row">
+                  <div class="check-status">
+                    <h3><%= t ".status" %></h3>
+                    <%= form_for :a, url: "#" do |f| %>
+                      <%= f.label :reading, class: "checkbox inline" do %>
+                        <%= f.check_box :reading_me %>
+                        <span><%= t(".reading")%></span>
```
Comment (mdmahmudurrahman) [inline app/views/details/info.html.erb]: remove optional ()

File: app/views/layouts/_footer.html.erb [hunk 13]
```diff
@@ -0,0 +1,37 @@
+<footer class="footer">
+  <div class="space-20"></div>
+  <div class="container">
+    <div class="row">
+      <div class="col-xs-12 col-sm-3 col-md-3">
+        <h4 class="text-white"><%= t ".social" %></h4>
+      </div>
+      <div class="col-xs-12 col-sm-3 col-md-3">
+        <h4 class="text-white"><%=link_to t(".about"), about_path %></h4>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_footer.html.erb]: Add a white space after the start of erb tag

File: app/views/shared/_error_messages.html.erb [hunk 14]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      <%= t ".form_contain" %><%= pluralize(@user.errors.count, t(".error")) %>.
```
Comment (mdmahmudurrahman) [inline app/views/shared/_error_messages.html.erb]: Please remove redundant ()

File: app/views/static_pages/about.html.erb [hunk 15]
```diff
@@ -0,0 +1,10 @@
+<% provide :title, t(".about") %>
+<!DOCTYPE html>
+<html>
+  <head>
+    <title><%= yield(:title) %><%= t ".book_app" %></title>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/about.html.erb]: Remove optional ()

File: app/views/static_pages/contact.html.erb [hunk 16]
```diff
@@ -0,0 +1,18 @@
+<% provide :title, t(".contact") %>
+<!DOCTYPE html>
+<html>
+  <head>
+    <title><%= yield(:title) %><%= t ".book_app" %></title>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/contact.html.erb]: The same here

File: app/views/static_pages/home.html.erb [hunk 17]
```diff
@@ -0,0 +1,12 @@
+<% provide :title, t(".home") %>
+<!DOCTYPE html>
+<html>
+  <head>
+    <title><%= full_title yield(:title) %></title>
+  </head>
+  <body>
+  	<%= render "layouts/header" %>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: Indentation

File: test/models/user_test.rb [hunk 18]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (mdmahmudurrahman) [inline test/models/user_test.rb]: Please remove all test files

**Comment by mdmahmudurrahman** [comment]
commented

File: app/views/details/info.html.erb (hunk 9 above)
Comment (thanhconguet214) [reply app/views/details/info.html.erb]: it's "form_for :status". I will create "status model" and use "form_for :status" to create or update status of a book user

**Comment by thanhconguet214** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 19]
```diff
@@ -0,0 +1,335 @@
...
+  float: left;
+  margin-right: 10px;
+}
+
+.gravatar_edit {
+  margin-top: 15px;
+}
+
+/*form*/
+#error_explanation {
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Add a new line after the comment

File: app/assets/stylesheets/custom.scss [hunk 20]
```diff
@@ -0,0 +1,335 @@
...
+      background: rgb(28, 184, 65); /* this is a green */
+    }
+    .button-read {
+      background: rgb(202, 60, 60); /* this is a maroon */
+    }
+    .button-waiting {
+      background: rgb(223, 117, 20); /* this is an orange */
+    }
+    .button-viewed {
+      background: rgb(66, 184, 221); /* this is a light blue */
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: 1. Remove the comment
2. Use color variable

File: app/assets/stylesheets/custom.scss [hunk 21]
```diff
@@ -0,0 +1,335 @@
...
+    display: inline-block;
+    width: 25%;
+  }
+  #description{
+    margin-top: 5%; 
+  }
+}
+
+/* Content */
+#content1{
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Add a line break after the comment

File: app/assets/stylesheets/custom.scss [hunk 22]
```diff
@@ -0,0 +1,335 @@
...
+  margin: 0 auto;
+  background-color: $color_content1;
+  #contfont2{
+    border-right: 1px dotted $color_black;
+  }
+}
+
+/* Search */
+#search{
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: The same here

File: app/views/layouts/_book_rating.html.erb [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+    <div class="col-xs-12 col-md-4 wow fadeInLeft" data-wow-delay = "0.1s" 
+      data-wow-offset ="50">
+      <div class="category-item well yellow" id="rating">
+        <div class="media">
+          <span class="badge badge-5">
+            <%= link_to t(".handico_office"), root_path %>
+          </span>
+          <div class="row" id="shared_by">
+            <div class="media-left col-md-6" id="img_update">
+             <%= image_link_to "img935-2-2.jpg", info_path %>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_book_rating.html.erb]: 2 space indentation

File: app/views/layouts/_book_read.html.erb [hunk 24]
```diff
@@ -0,0 +1,79 @@
...
+    <div class="col-xs-12 col-md-4 wow fadeInLeft" data-wow-delay = "0.1s" 
+      data-wow-offset ="50">
+      <div class="category-item well yellow" id="read">
+        <div class="media">
+          <span class="badge badge-5">
+            <%= link_to t(".handico_office"), root_path %>
+          </span>
+          <div class="row" id="shared_by">
+            <div class="media-left col-md-6" id="img_update">
+             <%= image_link_to "hanh-trinh-ve-phuong-dong.jpg", info_path %>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_book_read.html.erb]: 2 space indentation

File: app/views/layouts/_book_read.html.erb [hunk 25]
```diff
@@ -0,0 +1,79 @@
+<section id="section3" class = "gray-bg">
+  <div class="space-40"></div>
+  <div class="container section-book col-md-12 border-d6">
+    <div class="row">
+      <div class="col-xs-12 col-sm-8 col-sm-offset-2 col-md-6 
+        col-md-offset-3 text-center">
+        <h2 id="new_read"><strong><%= t ".book_read" %></strong></h2>
+        <div class="title-bar blue">
+        </div>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_book_read.html.erb]: Indentation

File: app/views/layouts/_book_read.html.erb [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+      </div>
+    </div>
+  </div>
+  <div class = "row">
+    <div class="col-md-12 border-d3 text-center">
+      <a class="btn btn-warning btn-xs uppercase" href="#"><%= t ".more" %></a>
+    </div>
+  </div>
+  <div class = "space-20"></div>
+  </div>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_book_read.html.erb]: Make the div indentation correct for this file

File: app/views/layouts/_book_update.html.erb [hunk 27]
```diff
@@ -0,0 +1,80 @@
...
+        </div>
+      </div>
+    </div>
+  </div>
+  <div class = "row">
+    <div class="col-md-12 border-d3 text-center">
+      <a class="btn btn-warning btn-xs uppercase" href="#"><%= t ".more" %></a>
+    </div>
+  </div>
+  <div class = "space-20"></div>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_book_update.html.erb]: Make the div indentation correct for this file

File: app/views/layouts/_book_viewed.html.erb [hunk 28]
```diff
@@ -0,0 +1,78 @@
...
+    <div class="col-xs-12 col-md-4 wow fadeInLeft" data-wow-delay = "0.1s" 
+      data-wow-offset ="50">
+      <div class="category-item well yellow" id="viewed">
+        <div class="media">
+          <span class="badge badge-5">
+            <%= link_to t(".handico_office"), root_path %>
+          </span>
+          <div class="row" id="shared_by">
+            <div class="media-left col-md-6" id="img_update">
+             <%= image_link_to "nxbtrestoryfull-27302013-093014", info_path %>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_book_viewed.html.erb]: Indentation

File: app/views/layouts/_book_waiting.html.erb [hunk 29]
```diff
@@ -0,0 +1,79 @@
...
+  <div class="container section-book col-md-12 border-d6">
+    <div class="row">
+      <div class="col-xs-12 col-sm-8 col-sm-offset-2 col-md-6 
+        col-md-offset-3 text-center">
+        <h2 id="new_waiting"><strong><%= t ".book_waiting" %></strong></h2>
+        <div class="title-bar blue">
+        </div>
+      </div>
+    </div>
+  <div class = "space-20"></div>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_book_waiting.html.erb]: Indentation

File: app/views/layouts/_search_book.html.erb [hunk 30]
```diff
@@ -0,0 +1,23 @@
...
+            </label>
+            <div class="search-result">
+              <ul id="data-search"></ul>
+            </div>
+          </div>
+        </strong>
+      </h2>
+      <div class="title-bar blue"></div>
+    </div>
+  <div class="space-20"></div>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_search_book.html.erb]: Indentation

File: app/views/static_pages/help.html.erb [hunk 31]
```diff
@@ -0,0 +1,12 @@
+<% provide :title, t(".help") %>
+<!DOCTYPE html>
+<html>
+  <head>
+    <title><%= yield :title %></title>
+  </head>
+  <body>
+    <h1>Help</h1>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/help.html.erb]: I18n

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by thanhconguet214** [comment]
ready

File: app/views/layouts/_shim.html.erb [hunk 32]
```diff
@@ -0,0 +1,5 @@
+<%= csrf_meta_tags %>
+<%= stylesheet_link_tag "custom", media: "all",
+  "data-turbolinks-track": "reload" %>
+<%= javascript_include_tag "custom", "data-turbolinks-track": "reload" %>
+<link rel=”stylesheet” href=”https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css”>
```
Comment (huongnguyenmta) [inline app/views/layouts/_shim.html.erb]: download thư viện về dùng đi em
không dùng cdn như này nhé

File: app/views/static_pages/about.html.erb [hunk 33]
```diff
@@ -0,0 +1,10 @@
+<% provide :title, t(".about") %>
+<!DOCTYPE html>
```
Comment (huongnguyenmta) [inline app/views/static_pages/about.html.erb]: mấy dòng này cần không thế em -_- (từ dòng 2-4)

**Comment by huongnguyenmta** [comment]
commented
(lengoi) chả chịu rì viu code trước khi gửi rì cả -_-

**Comment by thanhconguet214** [comment]
ready

File: app/views/static_pages/about.html.erb [hunk 34]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, t(".about") %>
+<head>
```
Comment (huongnguyenmta) [inline app/views/static_pages/about.html.erb]: mấy file này cần thẻ `head` với `body` để làm gì thế em??
Check những file tương tự nhé

**Comment by huongnguyenmta** [comment]
commented
(2tat)

**Comment by thanhconguet214** [comment]
ready

**Pull request merged**